### PR TITLE
TSK-1650: WorkbasketAccessMapper now handles KEY and DOMAIN in WHERE clause case insensitive

### PR DESCRIPTION
**Pull request by Tristan2357**
https://sonarcloud.io/dashboard?branch=TSK-1650&id=Tristan2357_taskana

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Review by holgerhagen — CHANGES_REQUESTED** [review]
Are we sure we do have enough tests (including all fields) to have a good feeling about the refactoring of the SQLs? I haven't checked that.

File: lib/taskana-core/src/test/java/acceptance/TaskanaEngineTestConfiguration.java [hunk 1]
```diff
@@ -174,9 +174,9 @@ private static DataSource createDefaultDataSource() {
 
     String jdbcDriver = "org.h2.Driver";
     String jdbcUrl =
-        "jdbc:h2:mem:taskana;IGNORECASE=TRUE;LOCK_MODE=0;"
-            + "INIT=CREATE SCHEMA IF NOT EXISTS TASKANA\\;"
-            + "SET COLLATION DEFAULT_de_DE ";
+        "jdbc:h2:mem:taskana;"
+            //  + "IGNORECASE=TRUE;"
```
[the remark this answers is not in the document]
Comment (Tristan2357) [reply lib/taskana-core/src/test/java/acceptance/TaskanaEngineTestConfiguration.java:178]: This could be a good question for our cop, if we want to keep our h2 case insensitive or not.

**Comment by Tristan2357** [comment]
As you can see from our sonarcloud analysis the new SqlProviders do have about 100% coverage and the mybatis scripts do not actually contain logic, except for db2 exceptions.
So i would say our tests are exhaustive enough in this case.

File: lib/taskana-core/src/test/java/acceptance/TaskanaEngineTestConfiguration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (holgerhagen) [reply lib/taskana-core/src/test/java/acceptance/TaskanaEngineTestConfiguration.java:178]: But you've removed it and added it to the CoP. This is fine and sufficient for me. From my point of view we could remove the comment and go ahead. What do you think?

**Review by holgerhagen — CHANGES_REQUESTED** [review]
We had some trouble with ambiguous column names lately. Please let us discuss this first before integrating
